src/jit/lowerxarch.cpp

```diff
@@ -2299,6 +2299,18 @@ void Lowering::TreeNodeInfoInitLogicalOp(GenTree* tree)
         // as reg optional.
         SetRegOptionalForBinOp(tree);
     }
+
+    if (varTypeIsByte(op1))
+    {
+        regMaskTP regMask;
+        bool      containedNode = (op2->gtLsraInfo.srcCount == 0) && (op2->gtLsraInfo.dstCount == 0);
+        if (!containedNode)
+        {
+            regMask = op2->gtLsraInfo.getSrcCandidates(l);
+            assert(regMask != RBM_NONE);
+            op2->gtLsraInfo.setSrcCandidates(l, regMask & ~RBM_NON_BYTE_REGS);
+        }
```

Comment:
Do we only care if it's a mem-op? Or would we want to constrain op1 if op2 is TYP_BYTE no matter what?